src/renderers/shared/fiber/ReactFiberCommitWork.js

```diff
@@ -495,6 +499,8 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
             if (__DEV__) {
               startPhaseTimer(finishedWork, 'componentDidMount');
             }
+            instance.props = finishedWork.memoizedProps;
+            instance.state = finishedWork.memoizedState;
```

Comment:
Not that I can think of, more concerned about the case I can't think of, like how I didn't consider error boundaries. Is this a perf concern?